src/foam/nanos/auth/UserAndGroupAuthService.java

```diff
@@ -225,6 +234,122 @@ public static String generateRandomSalt() {
     return saltBuilder.toString();
   }
 
+  //Used regex recommended here http://www.ex-parrot.com/pdw/Mail-RFC822-Address.html
+  public static boolean validateEmail(String email) {
```

Comment:
I suspect that this is very slow to compile every time. Add a TODO to benchmark and then reuse the pattern if benchmark warrants it.